### PR TITLE
DATA-817: Update zkevm dataflow etl

### PR DESCRIPTION
**Pull request by roopak-immutable**
# Description

- Update zkevm dataflow etl

## Jira

[DATA-817](https://immutable.atlassian.net/browse/DATA-817)

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation
- [ ] Configuration

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

[DATA-817]: https://immutable.atlassian.net/browse/DATA-817?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by tai-immutable — COMMENTED** [review]
@roopak-immutable Where does the jar come from?

**Comment by roopak-immutable** [comment]
> @roopak-immutable Where does the jar come from?

It's in the make file. 

https://github.com/immutable/im-data-blockchain-etl-dataflow/blob/master/Makefile#L41

File: target/blockchain-etl-dataflow-bundled-0.1.jar
Comment (jasadams-imtbl) [inline target/blockchain-etl-dataflow-bundled-0.1.jar:1]: I'm not accustomed to storing the binaries compiled from the source in source control, Is this deliberate?

File: terraform/main.tf [hunk 1]
```diff
@@ -1,3 +1,12 @@
+data "terraform_remote_state" "domain_prod_gcp_infra" {
+  backend   = "gcs"
+  workspace = terraform.workspace
+  config = {
+    bucket = "im-common-cicd-infra"
+    prefix = "terraform/state/im-data-gcp-department-infra"
+  }
+}
+
```
Comment (jasadams-imtbl) [inline terraform/main.tf]: I don't see any change relating to the above block being added, what is it for?

**Comment by tai-immutable** [comment]
Is there a way for us to make the jar compilation more repeatable?

I compiled using `make generate.jar`; the generated `.jar` md5 checksums are different (maybe expected, maybe not 🤷). I'm concerned that when I compiled it, my environment might have been different 😒

File: terraform/main.tf (hunk 1 above)
Comment (roopak-immutable) [reply terraform/main.tf]: Good pickup. Not sure how it came in there. Removed.

File: target/blockchain-etl-dataflow-bundled-0.1.jar
Comment (roopak-immutable) [reply target/blockchain-etl-dataflow-bundled-0.1.jar:1]: I have removed `.jar` file and included an actions step to generate the same.

Comment (tai-immutable) [reply target/blockchain-etl-dataflow-bundled-0.1.jar:1]: @roopak-immutable, I like this, thanks! Could we kill the binary from the PR?

**Review by jasadams-imtbl — APPROVED** [review]

File: target/blockchain-etl-dataflow-bundled-0.1.jar
Comment (roopak-immutable) [reply target/blockchain-etl-dataflow-bundled-0.1.jar:1]: Its already done @tai-immutable.

Check this branch https://github.com/immutable/im-data-blockchain-etl-dataflow/tree/d817

**Review by tai-immutable — APPROVED** [review]
LGTM

**Pull request merged**